samples/csharp/getting-started/Ranking_PersonalizedSort/PersonalizedRanking/Program.cs

```diff
@@ -0,0 +1,173 @@
+﻿using Microsoft.ML;
+using PersonalizedRanking.Common;
+using PersonalizedRanking.DataStructures;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+
+namespace PersonalizedRanking
+{
+    class Program
+    {
+        const string AssetsPath = @"../../../Assets";
+        const string TrainDatasetUrl = "https://aka.ms/mlnet-resources/benchmarks/MSLRWeb10KTrain720kRows.tsv";
+        const string TestDatasetUrl = "https://aka.ms/mlnet-resources/benchmarks/MSLRWeb10KTest240kRows.tsv";
```

Comment:
Doesn't hurt to list the validation dataset. This would let users experiment on the dataset by iterating using the scores on the validation set, then testing their final solution, exactly once, on the test dataset.

Generally the DS pattern for ML should look like:
1. Train on training set, get metrics from validation set
2. (iterate many times until happy with pipeline -- goto 1)
3. Train on found pipeline on combined train+validate set, get metrics on test set (exactly once) -- this is your final metrics
4. Retrain  pipeline on all data train+validate+test set. Send this newly created model to production.

The final estimate of how well your model will do in production is the metrics from step (3). The final model for production, trained on all available data, is trained at step (4).

Reply:
#Resolved